src/Hocon.Configuration/Config.cs

```diff
@@ -18,7 +18,7 @@ namespace Hocon
     ///     configuration string.
     /// </summary>
     [Serializable]
-    public class Config : HoconRoot, ISerializable
+    public class Config : HoconObject
```

Comment:
So `HoconObject` is now the base class we're using for `Config` - what's the significance of this change overall? I mostly just want to understand the direction you're taking things in.

For example:

> "`HoconRoot` turned out to be an unnecessary abstraction and exposing all of this functionality directly through `HoconObject` works just as well but even simpler with an overall smaller object graph"

That type of thing

Reply:
Correct, HoconRoot is unnecessary, it is practically a HoconObject. HoconRoot was there to pass around substitution list from included hocon files into its parent hocon files

Reply:
This will make the user facing Hocon DOM API a lot simpler, the user only have to deal with HoconObject, HoconArray, HoconLiteral, and Config class (HoconElement is an abstract class). We can always re-introduce HoconRoot, which will inherit HoconObject.

Reply:
Makes a lot of sense